From 2e452d98d66641dac37fb34787fdf2d7df4ed16a Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Tue, 28 Mar 2023 14:53:23 +0200
Subject: [PATCH] gmscompat: Spoof model, product and fingerprint with cust
 prop

Allows you to change the product, the model and the fingerprint during the Safetynet certification

use severals properties : persist.sys.phh.safetyspoof, persist.sys.phh.safetyspoof.model, etc...

Change-Id: I0c54b79fd97d0c7dd2eb57f981eee09adb3473b8
---
 .../internal/gmscompat/AttestationHooks.java  | 28 +++++++++++++++----
 1 file changed, 22 insertions(+), 6 deletions(-)

diff --git a/core/java/com/android/internal/gmscompat/AttestationHooks.java b/core/java/com/android/internal/gmscompat/AttestationHooks.java
index ef7a308a25bc..f4243c066f58 100644
--- a/core/java/com/android/internal/gmscompat/AttestationHooks.java
+++ b/core/java/com/android/internal/gmscompat/AttestationHooks.java
@@ -27,6 +27,20 @@ import java.util.Arrays;
 
 /** @hide */
 public final class AttestationHooks {
+
+
+            
+    private static final boolean SAFETYSPOOF_ENABLE =
+            android.os.SystemProperties.getBoolean("persist.sys.phh.safetyspoof", false);
+    private static final String SAFETYSPOOF_MODEL =
+            android.os.SystemProperties.get("persist.sys.phh.safetyspoof.model", "Pixel XL");
+    private static final String SAFETYSPOOF_DEVICE =
+            android.os.SystemProperties.get("persist.sys.phh.safetyspoof.device", "marlin");
+    private static final String SAFETYSPOOF_PRODUCT =
+            android.os.SystemProperties.get("persist.sys.phh.safetyspoof.product", "marlin");
+    private static final String SAFETYSPOOF_FINGERPRINT=
+            android.os.SystemProperties.get("persist.sys.phh.safetyspoof.fingerprint", "google/marlin/marlin:7.1.2/NJH47F/4146041:user/release-keys");
+                        
     private static final String TAG = "GmsCompat/Attestation";
 
     private static final String PACKAGE_GMS = "com.google.android.gms";
@@ -71,12 +85,14 @@ public final class AttestationHooks {
     }
 
     private static void spoofBuildGms() {
-        // Alter model name and fingerprint to avoid hardware attestation enforcement
-        setBuildField("FINGERPRINT", "google/marlin/marlin:7.1.2/NJH47F/4146041:user/release-keys");
-        setBuildField("PRODUCT", "marlin");
-        setBuildField("DEVICE", "marlin");
-        setBuildField("MODEL", "Pixel XL");
-        setVersionField("DEVICE_INITIAL_SDK_INT", Build.VERSION_CODES.N_MR1);
+        if (SAFETYSPOOF_ENABLE) {
+            // Alter model name and fingerprint to avoid hardware attestation enforcement
+            setBuildField("FINGERPRINT", SAFETYSPOOF_FINGERPRINT);
+            setBuildField("PRODUCT", SAFETYSPOOF_PRODUCT);
+            setBuildField("DEVICE", SAFETYSPOOF_DEVICE);
+            setBuildField("MODEL", SAFETYSPOOF_MODEL);
+            setVersionField("DEVICE_INITIAL_SDK_INT", Build.VERSION_CODES.N_MR1);
+        }
     }
 
     public static void initApplicationBeforeOnCreate(Application app) {
-- 
2.25.1

